apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.CreateCommonTestBucket/Program.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright 2020 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Auth.OAuth2;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Text;
+
+namespace Google.Cloud.Storage.V1.CreateCommonTestBucket
+{
+    /// <summary>
+    /// Creates and populates a bucket for cross-language testing purposes, where it's easy to do
+    /// "slightly the wrong thing".
+    /// </summary>
+    internal sealed class Program
+    {
+        private readonly string _bucket;
+        private readonly StorageClient _client;
+
+        private Program(string bucket)
+        {
+            _bucket = bucket;
+            _client = StorageClient.Create();
+        }
+
+        static int Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.WriteLine("Required arguments: <bucket>");
+                return 1;
+            }
+            var program = new Program(args[0]);
+            program.PopulateBucket();
+            return 0;
+        }
+
+        private void PopulateBucket()
+        {
+            MaybeCreateBucket();
+            DeleteExistingFiles();
+            CreateNormalizationFiles();
+            CreateGZippedContent();
+        }
+
+        private void MaybeCreateBucket()
+        {
+            try
+            {
+                _client.GetBucket(_bucket);
+                return;
+            }
+            catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+            {
+                // Need to create the bucket...
+                // (But let's do so outside the catch.)
+            }
+            Console.WriteLine($"Creating bucket '{_bucket}'");
+            var serviceAccount = GoogleCredential.GetApplicationDefault().UnderlyingCredential as ServiceAccountCredential;
+            if (serviceAccount is null)
+            {
+                throw new InvalidOperationException("Default credentials are not for a service account, so we can't determine the project ID");
```

Comment:
nit: Or take the projectId as an arg.
But really totally fine as is.

Reply:
I did wonder about that - it's basically an option if we find it's a problem :)